x/dex/keeper/end_block_place_orders.go

```diff
@@ -52,25 +62,37 @@ func (k *Keeper) HandleEBPlaceOrders(ctx context.Context, sdkCtx sdk.Context, tr
 	span.End()
 }
 
-func (k *Keeper) getPlaceSudoMsg(contractAddr string, registeredPairs []types.Pair) types.SudoOrderPlacementMsg {
+func (k *Keeper) getPlaceSudoMsg(contractAddr string, registeredPairs []types.Pair) []types.SudoOrderPlacementMsg {
+	contractDepositInfo := []types.ContractDepositInfo{}
+	for _, depositInfo := range k.DepositInfo[contractAddr].DepositInfoList {
+		contractDepositInfo = append(contractDepositInfo, dexcache.ToContractDepositInfo(depositInfo))
+	}
 	contractOrderPlacements := []types.ContractOrderPlacement{}
+	msgs := []types.SudoOrderPlacementMsg{
+		{
+			OrderPlacements: types.OrderPlacementMsgDetails{
+				Orders:   []types.ContractOrderPlacement{},
+				Deposits: contractDepositInfo,
+			},
+		},
+	}
 	for _, pair := range registeredPairs {
 		if orderPlacements, ok := k.OrderPlacements[contractAddr][pair.String()]; ok {
 			for _, orderPlacement := range orderPlacements.Orders {
 				if !orderPlacement.Liquidation {
 					contractOrderPlacements = append(contractOrderPlacements, dexcache.ToContractOrderPlacement(orderPlacement))
+					if len(contractOrderPlacements) == MAX_ORDERS_PER_SUDO_CALL {
```

Comment:
here if the number of contractOrderPlacements doesn't reach 50000 here, is the contractOrderPlacements discarded then?

Reply:
oh good catch

Reply:
can add a unit test for this one^